### PR TITLE
GUVNOR-2452: general improvements on artifact list in order to be eas…

### PR DESCRIPTION
**Pull request by porcelli**
…y to reuse (needed by new exec server ui)

### CODE REVIEW

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java [hunk 1]
```diff
@@ -53,42 +56,55 @@
...
+                    return row.getName();
+                }
+            };
+            nameColumn.setSortable( true );
+            nameColumn.setDataStoreName( JarListPageRequest.COLUMN_NAME );
+            addColumn( nameColumn,
+                       M2RepoEditorConstants.INSTANCE.Name() );
+        }
+
+        if ( columns.contains( ColumnType.GAV ) ) {
```
Comment (ederign) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: else if?

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java [hunk 2]
```diff
@@ -53,42 +56,55 @@
...
+                    return row.getGav().toString();
+                }
+            };
+            gavColumn.setSortable( true );
+            gavColumn.setDataStoreName( JarListPageRequest.COLUMN_GAV );
+            addColumn( gavColumn,
+                       M2RepoEditorConstants.INSTANCE.GAV() );
+        }
+
+        if ( columns.contains( ColumnType.LAST_MODIFIED ) ) {
```
Comment (ederign) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: else if?

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java (hunk 1 above)
Comment (porcelli) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: nope; we may have 0 to many columns

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java (hunk 2 above)
Comment (porcelli) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: nope

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java [hunk 3]
```diff
@@ -38,8 +39,9 @@
 
 import static org.junit.Assert.*;
 import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.*;
 
-@RunWith( MockitoJUnitRunner.class )
+@RunWith(MockitoJUnitRunner.class)
 public class ArtifactListTest {
 
     private static final int REQUEST_RANGE_LENGTH = 53;
```
Comment (ederign) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java]: just for the sake of my curiosity, why 53?

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java (hunk 1 above)
Comment (ederign) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/widgets/ArtifactListViewImpl.java]: k

**Comment by ederign** [comment]
+1

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java (hunk 3 above)
Comment (porcelli) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java]: ask original author ;) - i have no clue

Comment (ederign) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java]: https://upload.wikimedia.org/wikipedia/commons/8/80/Herbie-1138.jpg

**Comment by porcelli** [comment]
Found a bug, i'll update it in a few

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java (hunk 3 above)
Comment (Rikkola) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/test/java/org/guvnor/m2repo/client/widgets/ArtifactListTest.java]: The number is most likely based on the Stetson-Harrison method.

**Pull request closed without merging**